### PR TITLE
Minimize locking duration for decommitLock

### PR DESCRIPTION
**Pull request by cshung**

### CODE REVIEW

**Comment by Maoni0** [comment]
as discussed, this actually shouldn't be done since you'll hit a deadlock when multiple user threads call refresh.

**Comment by Maoni0** [comment]
ahh, I thought this was for a different problem where we talked about the lock order. I see this is actually about moving the lock to be around the verification, then yes, this is the right thing to do.

File: src/coreclr/gc/gc.cpp [hunk 1]
```diff
@@ -6067,7 +6067,7 @@ void gc_heap::release_segment (heap_segment* sg)
     FIRE_EVENT(GCFreeSegment_V1, heap_segment_mem(sg));
     size_t reserved_size = (uint8_t*)heap_segment_reserved (sg) - (uint8_t*)sg;
     reduce_committed_bytes (
-        sg,
```
Comment (Maoni0) [inline src/coreclr/gc/gc.cpp:6070]: nit - I see there are some white space changes, are those intentional?

Comment (cshung) [reply src/coreclr/gc/gc.cpp:6070]: Unintended, should not be there.

**Review by Maoni0 — APPROVED** [review]

**Pull request merged**